src/main/java/com/google/sps/servlets/ListEntriesServlet.java

```diff
@@ -38,8 +43,25 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+
+    String sessionId = request.getSession().getId();
+    Userinfo userInfo = null;
+    boolean isUserLoggedIn =
+        OAuthUtils.isUserLoggedIn(sessionId);
+
+    if (isUserLoggedIn) {
+      userInfo = OAuthUtils.getUserInfo(sessionId);
+    }
+    else {
+        response.sendRedirect("/login");
+    }
+
     Query<Entity> query =
-        Query.newEntityQueryBuilder().setKind("Entry").setOrderBy(OrderBy.desc("timestamp")).build();
+        Query.newEntityQueryBuilder()
+        .setKind("Entry")
+        // .setOrderBy(OrderBy.desc("timestamp"))
```

Comment:
Should we keep this line?

Reply:
i wanted to set the order by timestamp but wasn't able to figure out how to do it while filtering the entries at the same time so i left it commented out for a future update, but i'll just delete it for the time being.